### DISCUSSION TITLE
How to delete the memory?

### DISCUSSION THREAD

**Discussion by speakyourcode** [emscripten-core/emscripten#19135] [Q&A]
Hi guys!

Thank you for this awesome tool, I'm following the [Tutorial](https://emscripten.org/docs/getting_started/Tutorial.html#generating-html) to draw a square with SDL and all seems to work fine, but if I erase the square and paint it again, the memory keeps growing and haven't been able to figure out how to delete it. What am I missing?

I've set up the following [Sandbox](https://codesandbox.io/p/github/speakyourcode/emscripten-example/main?file=README.md&workspace=%257B%2522activeFilepath%2522%253A%2522%252Fhtml%252Findex.html%2522%252C%2522openFiles%2522%253A%255B%2522README.md%2522%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522spaces%2522%253A%257B%2522clg3qsggf000x2v6ir38jt8w6%2522%253A%257B%2522key%2522%253A%2522clg3qsggf000x2v6ir38jt8w6%2522%252C%2522name%2522%253A%2522Default%2522%252C%2522devtools%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522npx%2520http-server%2520-p%25205514%2520html%2522%252C%2522port%2522%253A5514%252C%2522key%2522%253A%2522clg3qv4ix00gy2v6if8bagk8x%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522npx%2520http-server%2520-p%25205514%2520html%2522%252C%2522key%2522%253A%2522clg3quvux00aw2v6ibu9h1eow%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D%257D%252C%2522currentSpace%2522%253A%2522clg3qsggf000x2v6ir38jt8w6%2522%252C%2522spacesOrder%2522%253A%255B%2522clg3qsggf000x2v6ir38jt8w6%2522%255D%252C%2522hideCodeEditor%2522%253Afalse%257D) with two buttons: paint and erase. 

You can see how the files were generated in the README, I essentially use MODULARIZE for web ENVIRONMENT.

With each paint, the memory grows about 20MB and haven't been able to delete it. You can see in the Memory Browser Tools that the memory is there even if I delete the canvas.

Any ideas? Thanks

**Comment by kripken** [comment]
How are you measuring memory? Taking heap snapshots in Chrome devtools, I see the page uses around 41.5 MB both at startup and after many presses of "paint".

**Reply by speakyourcode** [reply]
Yes, taking heap snapshots in the devtools, it increases around 20MB with each paint

**Reply by kripken** [reply]
On which browser?

**Reply by speakyourcode** [reply]
for some reason it doesn't seem to happen in the Sandbox, but as soon as I serve the page locally I get the following

![image](https://user-images.githubusercontent.com/58657886/230150013-f25a5419-6259-4b93-be1e-ae6c14707cd1.png)

snapshot 1) page load
snapshot 2) paint, erase
snapshot 3) paint, erase (again)

I've tested it from Chrome and Edge in Windows, and from Chrome in Mac, same result

**Reply by speakyourcode** [reply]
Any ideas? were you able to reproduce it?
